protocol/0075-PLUP_protocol_upgrades.md

```diff
@@ -0,0 +1,167 @@
+
+NB: by the time of the block height of the upgrade the validators must have downloaded and built the binaries and prepared them in the right location as required by the vega-visor.
+
+## Framework / data structures
+
+As protobuf:
+```go
+message ProtocolUpgradeProposal {
+   // The block height at which to perform the upgrade
+   uint64 upgrade_block_height = 1;
+   // the release tag for the vega binary
+   string vega_release_tag = 2;
+   // the release tag for the data-node binary
+   string data_node_release_tag = 3;
+}
+
+enum ProtocolUpgradeProposalStatus {
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_UNSPECIFIED = 0;
+  // The proposal is pending
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_PENDING = 1;
+  // The proposal is approved
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_APPROVED = 2;
+  // The proposal is rejected
+  PROTOCOL_UPGRADE_PROPOSAL_STATUS_REJECTED = 3;
+}
+
+message ProtocolUpgradeEvent {
+   // The block height at which to perform the upgrade
+   uint64 upgrade_block_height = 1;
+   // the release tag for the vega binary
+   string vega_release_tag = 2;
+   // the release tag for the data-node binary
+   string data_node_release_tag = 3;
+   // tendermint validators that have agreed to the upgrade
+  repeated string approvers = 4;
+  // the status of the proposal
+  ProtocolUpgradeProposalStatus status = 5;
+}
+
+```
+
+## Acceptance criteria
+
+### Invalid proposals - Rejections (<a name="0075-PLUP-001" href="#0075-PLUP-001">0075-PLUP-001</a>)
+   - A network with 5 validators 
+   - Validator 1 proposes a protocol upgrade to an invalid tag (https://semver.org/) - should result in an error
+   - Validator 2 proposes a protocol upgrade on a block height preceding the current block - should result in an error
+   - Propose and enact a version downgrade 
+   - Non-validator attempts to propose upgrade
+   - Ersatz validator attempts to propose upgrade
+ 
+### Block height validation (<a name="0075-PLUP-002" href="#0075-PLUP-002">0075-PLUP-002</a>)
+   Proposal will not be accepted as valid if validator
+   -  Proposes an upgrade block in the past - should be rejected / failed check - sign via wallet / check wallet
+   -  Proposes a negative upgrade block
+   -  Proposes a 0 upgrade block
+   -  Proposes (string/other upgrade block)
+   -  Proposes with supply a block height
+
+### VISOR (<a name="0075-PLUP-005" href="#0075-PLUP-005">0075-PLUP-005</a>) 
+   - Can be seen to automatically download the latest version for install when available at the source location when file meets the format criteria defined
+   - Visor automatically upgrades validators to proposed version if requiredMajority has been reached
+   
+### Epochs (<a name="0075-PLUP-006" href="#0075-PLUP-006">0075-PLUP-006</a>) 
+   - Proposing an upgrade block which ought to be the end of an epoch
+   - Propose an upgrade block which should result in a new network in the same epoch
```

Comment:
These are maybe 3 different test cases, which one do we mark with the AC? Similar deal for several of the above... Our current system seems to assume one python test case per AC... For _some_ of these I could theoretically cram them into one case, but for some of the others it doesn't necessarily make sense.